src/json/json.cc

```diff
@@ -258,6 +258,130 @@ STATIC JsonUtilCode parseSetCmdArgs(ValkeyModuleString **argv, const int argc, S
+        args_list[i].path = ValkeyModule_StringPtrLen(argv[i * 3 + 2], nullptr);
+        args_list[i].json = ValkeyModule_StringPtrLen(argv[i * 3 + 3], &args_list[i].json_len);
+        int type = ValkeyModule_KeyType(args_list[i].key);
+        if (type != VALKEYMODULE_KEYTYPE_EMPTY && ValkeyModule_ModuleTypeGetType(args_list[i].key) != DocumentType) {
+            FreeMSetCmdArgs(ctx, args_list, num_keys);
+            return JSONUTIL_NOT_A_DOCUMENT_KEY;
+        }
+
+        bool is_new_valkey_key = (type == VALKEYMODULE_KEYTYPE_EMPTY);
+        args_list[i].is_root_path = jsonutil_is_root_path(args_list[i].path);
+
+        if (is_new_valkey_key && !args_list[i].is_root_path) {
+            FreeMSetCmdArgs(ctx, args_list, num_keys);
+            return JSONUTIL_COMMAND_SYNTAX_ERROR;
+        }
+        JDocument *doc;
+        // begin tracking memory
+        int64_t begin_val = jsonstats_begin_track_mem();
+
+        if (args_list[i].is_root_path) {
+            rc = dom_parse(ctx, args_list[i].json, args_list[i].json_len, &doc);
+            if (rc != JSONUTIL_SUCCESS) {
+                FreeMSetCmdArgs(ctx, args_list, num_keys);
+                return rc;
+            }
+
+            // end tracking memory
+            int64_t delta = jsonstats_end_track_mem(begin_val);
+            args_list[i].doc_size = dom_get_doc_size(doc) + delta;
+
+            if (json_is_instrument_enabled_insert() || json_is_instrument_enabled_update()) {
+                size_t len;
+                const char* key_cstr = ValkeyModule_StringPtrLen(args_list[i].key_str,&len);
+                std::size_t key_hash = std::hash<std::string_view>{}(std::string_view(key_cstr, len));
+                ValkeyModule_Log(ctx, "warning",
+                                "Dump document structure before setting JSON key (hashed) %zu whole doc %p:",
+                                key_hash, static_cast<void *>(doc));
+                DumpRedactedJValue(doc->GetJValue(), nullptr, "warning");
+            }
+        } else {
+            doc = static_cast<JDocument*>(ValkeyModule_ModuleTypeGetValue(args_list[i].key));
+            if (doc == nullptr) {
+                FreeMSetCmdArgs(ctx, args_list, num_keys);
+                // return ValkeyModule_ReplyWithError(ctx, ERRMSG_JSON_DOCUMENT_NOT_FOUND);
+                return JSONUTIL_DOCUMENT_KEY_NOT_FOUND;
+            }
+            rc = dom_verify_value(ctx, doc, args_list[i].path, args_list[i].json);
+            if (rc != JSONUTIL_SUCCESS) {
+                FreeMSetCmdArgs(ctx, args_list, num_keys);
+                return rc;
+            }
+            // end tracking memory
+            int64_t delta = jsonstats_end_track_mem(begin_val);
+            args_list[i].doc_size = dom_get_doc_size(doc) + delta;
+        }
+    }
+
+    memset(args, 0, (num_keys * sizeof(MSetCmdArgs)) + sizeof(size_t));
```

Comment:
This is terribly wrong, causing memory corruption. This is the root cause of the SEGV. Change it to:
```
memset(args, 0, sizeof(MSetCmdArgsList));
```
And we'd better move the line to the beginning of the function (not required but as a good practice).